storage/buffer.go

```diff
@@ -672,7 +679,12 @@ func addH(s hSample, buf []hSample, r *sampleRing) []hSample {
 		}
 	}
 
-	buf[r.i] = s
+	buf[r.i].t = s.t
+	if buf[r.i].h == nil {
+		buf[r.i].h = s.h.Copy()
```

Comment:
Same question as above, these copies are new, do we need them?

Reply:
This copy is needed because we now use a single histogram object for decoding chunks (the `hReader` and `fhReader` variables), so we have to copy those samples when we store them in the buffer. Otherwise, the next read will overwrite buffered samples.

Reply:
I see, does that mean we're copying twice? Once in hReader from the chunk, then copy again into the sample buffer?

(probably a next step to optimize away if yes)

Reply:
I see, took me a while to understand how hReader is used, as it's not directly used to read from